docs/upgrading.rst

```diff
@@ -0,0 +1,447 @@
+Upgrading to Newer Releases
+===========================
+
+This section documents migration paths to new releases.
+
+Upgrading to 6.0
+++++++++++++++++
+
+In webargs 6, the way that data is loaded from various locations and passed to
+your schemas changed significantly. This may require you to change several
+different parts of your webargs usage.
+
```

Comment:
I tried to reword this, but in fact, I'd just remove it.

Next paragraphs speak for themselves.

Reply:
👍 I'll remove it in a fixup commit.